.github/workflows/gitlab-push.yml

```diff
@@ -16,6 +16,37 @@ env:
   PR_NUMBER: ${{ github.event.pull_request.number }}
 
 jobs:
+  push_to_gitlab_improved:
+    name: Push To GitLab (Improved)
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Push to GitLab
+        run: |
+          set -eux
+          URL="https://push-from-github:${{ secrets.PUSH_TO_GITLAB_TOKEN }}@gitlab.com/dfinity-lab/public/ic.git"
+
+          if git remote get-url gitlab >/dev/null 2>&1; then
+              git remote set-url gitlab "$URL"
+          else
+              git remote add gitlab "$URL"
+          fi
+
+          BRANCH="mirroring-improved-${BRANCH_NAME}"
+          git fetch origin mirroring
+          git checkout -b "$BRANCH"
+          git cherry-pick "$(git rev-parse origin/mirroring)"
```

Comment:
why does `git rev-parse` need to be used? would `git cherry-pick origin/mirroring` not work?